doc/user/content/manage/troubleshooting.md

```diff
@@ -10,77 +10,309 @@ aliases:
+
+```sql
+EXPLAIN 
+SELECT auctions.item, avg(bids.amount) AS average_bid
+FROM bids
+JOIN auctions ON bids.auction_id = auctions.id
+WHERE bids.bid_time < auctions.end_time
+GROUP BY auctions.item
+```
+```
+                                        Optimized Plan
+-----------------------------------------------------------------------------------------------
+ Explained Query:                                                                             +
+   Project (#0, #3)                                                                           +
+     Map ((bigint_to_double(#1) / bigint_to_double(case when (#2 = 0) then null else #2 end)))+
+       Reduce group_by=[#1] aggregates=[sum(#0), count(*)]                                    +
+         Project (#1, #4)                                                                     +
+           Filter (#2 < #5)                                                                   +
+             Join on=(#0 = #3) type=differential                                              +
+               ArrangeBy keys=[[#0]]                                                          +
+                 Project (#2..=#4)                                                            +
+                   Get materialize.qck.bids                                                   +
+               ArrangeBy keys=[[#0]]                                                          +
+                 Project (#0, #2, #3)                                                         +
+                   Get materialize.qck.auctions                                               +
+
+(1 row)
+```
 
-## Limitations
+The plan describes the specific Differential Dataflow operators that are used to evaluate the query.
+Some of these operators resemble relational algebra or map reduce style operators (`Filter`, `Join on`, `Map`). 
+Others are specific to differential dataflow (`Get`, `ArrangeBy`).
 
+In general, a high level understanding of what these operators do is sufficient for effective debugging:
+`Filter` filters records, `Join on` joins records from two or more inputs, `Map` applies a function to transform records, etc.
+The `ArrangeBy` operator indexes data and stores it in memory for fast point lookups.
+It's not important to have a deep understanding of all these operators for effective debugging, though.
+
+Behind the scenes, the operator graph is turned into a dataflow that can be evaluated by the Differential Dataflow runtime.
+The dataflow is organized in a hierarchical structure. 
+It contains operators and so-called regions, which in turn contain operators and (sub)regions.
+
+In our example, the dataflow contains a InputRegion and a BuildRegion.
+And the BuildRegion contains the operators from the above plan.
+
+![Regions and operator visualization](/images/regions-and-operators.png)
```

Comment:
If we're using this image, we should clearly tell users how to access the memory visualizer.